packages/babel-preset-env/src/polyfills/corejs3/entry-plugin.js

```diff
@@ -21,6 +21,9 @@ function isBabelPolyfillSource(source) {
 }
 
 function isCoreJSSource(source) {
+  if (typeof source === "string") {
+    source = source.replace(/(\/(index)?)?(\.js)?$/i, "").toLowerCase();
```

Comment:
```suggestion
    source = source.replace(/([\\/](index)?)?(\.js)?$/i, "").toLowerCase();
```

We need to also handle windows paths.

Reply:
Makes sense 👍